sdk/python/kfp/compiler/compiler.py

```diff
@@ -332,6 +332,36 @@ def _build_conventional_artifact(name):
     if op.num_retries:
       template['retryStrategy'] = {'limit': op.num_retries}
 
+    if op.sidecars:
+      def _sidecar_to_template(tsidecar):
```

Comment:
Can I ask if you can extract the code to be a common utility to convert a Container to k8s container json spec and reuse the method for both main container and sidecars? Thanks.